Dynatrace/dynatrace-operator
# Propagate annotations from namespace to pods for metadata enrichment

## Description

In a previous PR we already fetch custom rules that define which namespace annotations and labels should be copied to the pod. This PR actually copies the metadata.

[Jira](https://dt-rnd.atlassian.net/browse/K8S-9942)


## How can this be tested?

- Setup in the same way as in #3327. 
- Add labels to the namespace
- Create dynakube with enabled metadata enrichment
- Create test pod and check if metadata enrichment annotations exist
